src/main/java/com/company/slidingwindow/SubArrayProductLessThanK.java

```diff
@@ -0,0 +1,40 @@
+package com.company.slidingwindow;
+
+public class SubArrayProductLessThanK {
+    public int numSubarrayProductLessThanK(int[] nums, int k) {
+        int product = 1;
+        int left = 0;
+        int subArrayCount = 0;
+        for (int right = 0; right < nums.length; right++) {
+            product *= nums[right];
+            while (product >= k) {
+                product /= nums[left];
```

Comment:
Check for the problem constraints if the given array values will be always positive or non zero value. If not, it may throw a runtime exception. 

If the constraint is given as positive or non zero, you can close this conversation and merge the PR.

Reply:
Constraints:

1 <= nums.length <= 3 * 104
1 <= nums[i] <= 1000
0 <= k <= 106

Based on the contraints, the array values are positive